openmmtools/states.py

```diff
@@ -1235,9 +1241,12 @@ def _initialize(self, system, temperature=None, pressure=None):
         # If pressure is None, we try to infer the pressure from the barostat.
         barostat = self._find_barostat(system)
         if pressure is None and barostat is not None:
-            self._pressure = barostat.getDefaultPressure()
+            self._pressure = self._get_barostat_pressure(barostat)
         else:
             self._pressure = pressure  # Pressure here can also be None.
+        self._barostat_type = openmm.MonteCarloBarostat
```

Comment:
What do you think about making `_barostat_type` a private property that simply returns `type(self.barostat)`? Because this attribute is accessed very rarely, I'd prefer to make it slightly slower but avoid redundancies.

Reply:
I would rather like to keep it as it is. The reason is that `_initialize` calls `_is_barostat_consistent` indirectly (through `_unsafe_set_system` and `_check_system_consistency`_). At this point, `self.barostat` is not available, since `self._standard_system` is not initialized, yet.

Reply:
I see what you mean. It may be safe after all though because, if I understand correctly the code I wrote, `_is_barostat_consistent` shouldn't be called when `self.barostat` is unavailable (I'm looking at these lines: https://github.com/Olllom/openmmtools/blob/other_barostats/openmmtools/states.py#L1341-L1342).